src/clj/fluree/db/query/api.cljc

```diff
@@ -204,33 +204,33 @@
         (dataset db-map defaults)))))
 
 (defn query-connection-fql
-  [conn query did]
+  [conn query opts]
   (go-try
-   (let [{:keys [t opts] :as query*} (-> query
-                                         syntax/coerce-query
-                                         (update :opts sanitize-query-options did))
-         default-aliases (some-> query* :from util/sequential)
-         named-aliases   (some-> query* :from-named util/sequential)]
-     (if (or (seq default-aliases)
-             (seq named-aliases))
-       (let [ds       (<? (load-dataset conn default-aliases named-aliases t opts))
-             query**  (update query* :opts dissoc :meta :max-fuel ::util/track-fuel?)
-             max-fuel (:max-fuel opts)]
-         (if (::util/track-fuel? opts)
-           (<? (track-query ds max-fuel query**))
-           (<? (fql/query ds query**))))
-       (throw (ex-info "Missing ledger specification in connection query"
-                       {:status 400, :error :db/invalid-query}))))))
-
+    (let [{:keys [t opts] :as sanitized-query} (-> query
```

Comment:
This appears to overwrite the `opts` function argument... so it is unused.

Reply:
Actually I see (:did opts) is pulled from it before overwriting.

If only `:did` is being used, then it is far more clear to have `did` as a function argument than an opaque `opts` map what we don't know what is supposed to be inside it.